addons/mrp/models/stock_scrap.py

```diff
@@ -34,3 +34,30 @@ def _prepare_move_values(self):
             else:
                 vals.update({'raw_material_production_id': self.production_id.id})
         return vals
+
+    @api.onchange('lot_id')
+    def _onchange_serial_number(self):
+        if self.product_id.tracking == 'serial' and self.lot_id and self.production_id:
+            quants = self.env['stock.quant'].search([('product_id', '=', self.product_id.id),
+                                                     ('lot_id', '=', self.lot_id.id),
+                                                     ('quantity', '!=', 0),
+                                                     ('company_id', '=', self.company_id.id),
+                                                     ('location_id.usage', 'in', ('internal', 'transit'))])
+            sn_locations = quants.mapped('location_id')
+            if sn_locations and self.location_id not in sn_locations:
+                recommended_location = self.env['stock.location']
+                for location in sn_locations:
+                    if location == self.production_id.location_dest_id or location in self.production_id.location_dest_id.child_ids:
```

Comment:
child_ids contains only the first level of hierarchy. 
use this instead
```python
if self.production_id.location_dest_id.parent_path in location.parent_path
```

Reply:
Ah dang, switched it.